### PR TITLE
Separate DockerManager and Hypervisor classes (base: #3203)

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
Follow up on #3203

### CODE REVIEW

**Review by maaktweluit — APPROVED** [review]
Nice cleanup!

LGTM

File: golem/docker/manager.py [hunk 1]
```diff
@@ -870,21 +864,16 @@ def _config(self, name):
...
-        'driver': 'com.docker.driver',
-        'hyperkit': 'com.docker.hyperkit',
-        'vpnkit': 'com.docker.vpnkit',
-    }
+    PROCESSES = dict(
+        app=f'{APP}/Contents/MacOS/Docker',
+        driver='com.docker.driver',
+        hyperkit='com.docker.hyperkit',
+        vpnkit='com.docker.vpnkit',
+    )
```
Comment (Wiezzel) [inline golem/docker/manager.py]: Why?

File: golem/docker/manager.py [hunk 2]
```diff
@@ -331,21 +338,30 @@ def setup(self) -> None:
...
-    def create(self, name: Optional[str] = None, **params):
-        raise NotImplementedError
+    @classmethod
+    def _new_instance(
+            cls,
+            get_config_fn: GetConfigFunction,
+            docker_vm: str = DOCKER_VM_NAME) -> Optional['Hypervisor']:
+        return cls(get_config_fn, docker_vm=docker_vm)
+
+    def command(self, *args, **kwargs) -> Optional[str]:
```
Comment (Wiezzel) [inline golem/docker/manager.py]: Why not classmethod?

File: golem/docker/manager.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/docker/manager.py]: I avoid using string literals / quotes anywhere I can.

**Pull request merged**